test/games/strategy/triplea/delegate/BattleCalculatorTest.java

```diff
@@ -139,18 +144,24 @@ public void testAACasualtiesLowLuckMixedWithChooseAACasualties() {
     final Collection<Unit> planes = bomber(data).create(6, british(data));
     planes.addAll(fighter(data).create(6, british(data)));
     final Collection<Unit> defendingAA = territory("Germany", data).getUnits().getMatches(Matches.UnitIsAAforAnything);
-    m_bridge.setRemote(new DummyTripleAPlayer() {
-      @Override
-      public CasualtyDetails selectCasualties(final Collection<Unit> selectFrom,
-          final Map<Unit, Collection<Unit>> dependents, final int count, final String message, final DiceRoll dice,
-          final PlayerID hit, final Collection<Unit> friendlyUnits, final PlayerID enemyPlayer,
-          final Collection<Unit> enemyUnits, final boolean amphibious, final Collection<Unit> amphibiousLandAttackers,
-          final CasualtyList defaultCasualties, final GUID battleID, final Territory battlesite,
-          final boolean allowMultipleHitsPerUnit) {
-        final List<Unit> selected = Match.getNMatches(selectFrom, count, Matches.UnitIsStrategicBomber);
-        return new CasualtyDetails(selected, new ArrayList<>(), false);
-      }
-    });
+    TripleAPlayer player = MockObjects.getDummyPlayer();
+    when(player.selectCasualties(any(Collection.class), any(Map.class), any(int.class), any(String.class),
+        any(DiceRoll.class), any(PlayerID.class), any(Collection.class), any(PlayerID.class), any(Collection.class),
+        any(boolean.class), any(Collection.class),
+        any(CasualtyList.class), any(GUID.class), any(Territory.class), any(boolean.class)))
+            .thenAnswer(new Answer<CasualtyDetails>() {
```

Comment:
`thenReturn`is preferred, `thenAnswer` is a bit of a last resort. 
In this case, the parameter value can likely be known, in which case we can expect an exact value, and return an exact value that we have constructed in the `@Setup`